src/test/java/com/hazelcast/commandline/member/MemberCommandLineTest.java

```diff
@@ -0,0 +1,193 @@
+package com.hazelcast.commandline.member;
+
+import com.hazelcast.commandline.CommandLineTestSupport;
+import com.hazelcast.commandline.HazelcastCommandLine;
+import com.hazelcast.core.LifecycleEvent;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.io.*;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+import static junit.framework.TestCase.assertEquals;
+import static junit.framework.TestCase.assertTrue;
+
+public class MemberCommandLineTest extends CommandLineTestSupport {
+
+    private MemberCommandLine memberCommandLine;
+
+    @Before
+    public void setup() {
+        resetOut();
+        memberCommandLine = new MemberCommandLine(out, err);
+    }
+
+    @After
+    public void close() throws IOException, InterruptedException {
+        killAllRunningHazelcastInstances();
+        removeFiles();
+    }
+
+    private void removeFiles() throws IOException {
+        Path pathToBeDeleted = Files.createDirectories(Paths.get(HazelcastCommandLine.HAZELCAST_HOME));
+
+        Files.walk(pathToBeDeleted)
+                .sorted(Comparator.reverseOrder())
+                .map(Path::toFile)
+                .forEach(File::delete);
+    }
+
+    private void killAllRunningHazelcastInstances() throws IOException, InterruptedException {
+        String out = getRunningJavaProcesses();
+        List<String> pids = getHazelcastInstancePids(out);
+        runCommand("/bin/kill -9 " + String.join(" ", pids));
+
+    }
+
+    private List<String> getHazelcastInstancePids(String out) {
+        List<String> pids = new ArrayList<>();
+        for (String line : out.split("\n")) {
+            if (line.contains(HazelcastMember.class.getSimpleName())) {
+                pids.add(line.split(" ")[0]);
+            }
+        }
+        return pids;
+    }
+
+    @Test(timeout = 10000)
+    public void test_start() throws IOException, ClassNotFoundException, InterruptedException {
+        memberCommandLine.start(null, null, null, null,
+                false, null, null);
+        assertTrue(memberCommandLine.getProcessOutput().anyMatch(out ->
+                out.contains(LifecycleEvent.LifecycleState.STARTED.toString())));
+    }
+
+    @Test(timeout = 10000)
+    public void test_start_withConfigFile() throws IOException, ClassNotFoundException, InterruptedException {
```

Comment:
The structure of each of your tests seem chaotic. The test should have a clear structure: given, when, then. You can make if visible by explicitly using comments: `// given // when // then` or just by putting empty lines between each section. Here it's even worse, because you specify `groupName` (which is part of `// then`), next you have `// when`, and again `// then`.

Reply:
I mostly follow the `// given // when // then` strategy in my unit tests, just I don't prefer to use comments or empty lines between each section. I can use them for ease of reading. But I didn't understand why you think `groupName` is a part of `// then`? It is given in the beginning and placed in the config file. I just don't extract it from the file again.